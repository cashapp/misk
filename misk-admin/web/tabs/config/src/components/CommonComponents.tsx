/** @jsx jsx */
import { css } from "@emotion/core"

export const cssFloatLeft = css`
  float: left;
  margin: 5px 10px 0 0;
`

export const cssFloatRight = css`
  float: right;
  margin: 5px 0 0 10px;
`

export const cssTableScroll = css`
  height: 100%;
  width: 100%;
  overflow-x: auto;
  overflow-y: auto;
  white-space: nowrap;
`

export const cssHeader = css`
  display: flex;
  justify-content: space-between;
`
