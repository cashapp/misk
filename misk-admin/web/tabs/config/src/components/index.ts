import ConfigComponent from "./ConfigComponent"
export { ConfigComponent }
