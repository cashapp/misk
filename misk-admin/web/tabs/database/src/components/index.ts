export * from "./QuantityButton"
export * from "./CommonComponents"
export * from "./QueryResultTableComponent"
export * from "./SkeletonWebActionsComponent"
export * from "./StatusTagComponent"
