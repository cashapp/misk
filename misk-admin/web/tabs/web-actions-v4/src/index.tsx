import React, { useState } from "react"

import { createRoot } from "react-dom/client"
import RequestEditor from "@misk-console/ui/RequestEditor"
import { ChakraProvider, HStack, VStack } from "@chakra-ui/react"
import ResponseViewer from "@misk-console/ui/ResponseViewer"
import "ace-builds"
import "ace-builds/webpack-resolver"

function App() {
  const [response, setResponse] = useState<string | null>(null)
  return (
    <VStack height="100vh" spacing={0} bg="gray.600" alignItems="start">
      <HStack bg="gray.200" spacing={2} p={2} flexGrow={1} width="100%">
        <RequestEditor onResponse={setResponse} />
        <ResponseViewer toRender={response} />
      </HStack>
    </VStack>
  )
}

const container = document.getElementById("root")

createRoot(container!).render(
  <ChakraProvider>
    <App />
  </ChakraProvider>
)
