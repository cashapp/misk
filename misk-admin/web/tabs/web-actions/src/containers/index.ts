import FilterWebActionsContainer from "./FilterWebActionsContainer"
import RequestFormContainer from "./RequestFormContainer"
import RequestFormFieldBuilderContainer from "./RequestFormFieldBuilderContainer"
import SendRequestCollapseContainer from "./SendRequestCollapseContainer"
import TabContainer from "./TabContainer"
import WebActionCardContainer from "./WebActionCardContainer"
import WebActionsContainer from "./WebActionsContainer"
export {
  FilterWebActionsContainer,
  RequestFormContainer,
  RequestFormFieldBuilderContainer,
  SendRequestCollapseContainer,
  TabContainer,
  WebActionCardContainer,
  WebActionsContainer
}
