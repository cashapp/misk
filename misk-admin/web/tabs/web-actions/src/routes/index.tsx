import * as React from "react"
import { Route, Switch } from "react-router"
import { TabContainer as OldTabContainer } from "../containers"
import TabContainer from "../rewrite/TabContainer"

const routes = (
  <div>
    <Switch>
      <Route path="/_admin/web-actions/" component={TabContainer} />
      <Route path="/_admin/web-actions-old/" component={OldTabContainer} />
      {/* Do not include a Route without a path or it will display during on all tabs */}
    </Switch>
  </div>
)

export default routes
