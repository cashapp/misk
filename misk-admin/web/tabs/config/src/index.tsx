import { createApp, createIndex } from "@misk/core"
import * as Ducks from "./ducks"
import routes from "./routes"
export * from "./components"
export * from "./containers"

createIndex("config", createApp(routes), Ducks)
