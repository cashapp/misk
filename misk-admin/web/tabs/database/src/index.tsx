import { createApp, createIndex } from "@misk/core"
import * as Ducks from "./ducks"
import routes from "./routes"
export * from "./containers"

createIndex("database", createApp(routes), Ducks)
