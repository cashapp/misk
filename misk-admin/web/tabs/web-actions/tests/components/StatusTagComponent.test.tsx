import React from "react"
import { cleanup, render } from "@testing-library/react"
import { StatusTagComponent } from "../../src/components"

describe("StatusTagComponent", () => {
  afterEach(cleanup)
  it("StatusTagComponent renders empty when no status given", () => {
    const { asFragment } = render(
      <StatusTagComponent status={0} statusText={""} />
    )
    expect(asFragment()).toMatchSnapshot()
  })
  it("StatusTagComponent renders status 200 ok intent primary", () => {
    const { asFragment } = render(
      <StatusTagComponent status={200} statusText={"OK"} />
    )
    expect(asFragment()).toMatchSnapshot()
  })
  it("StatusTagComponent renders status 404 not found intent warning", () => {
    const { asFragment } = render(
      <StatusTagComponent status={404} statusText={"NOT FOUND"} />
    )
    expect(asFragment()).toMatchSnapshot()
  })
  it("StatusTagComponent renders status 500 internal error intent danger", () => {
    const { asFragment } = render(
      <StatusTagComponent status={500} statusText={"INTERNAL ERROR"} />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
