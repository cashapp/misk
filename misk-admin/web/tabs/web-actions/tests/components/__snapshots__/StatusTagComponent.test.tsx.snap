// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusTagComponent StatusTagComponent renders empty when no status given 1`] = `
<DocumentFragment>
  <span />
</DocumentFragment>
`;

exports[`StatusTagComponent StatusTagComponent renders status 200 ok intent primary 1`] = `
<DocumentFragment>
  <span
    class="bp3-tag bp3-intent-success"
  >
    <span
      class="bp3-text-overflow-ellipsis bp3-fill"
    >
      200 OK
    </span>
  </span>
</DocumentFragment>
`;

exports[`StatusTagComponent StatusTagComponent renders status 404 not found intent warning 1`] = `
<DocumentFragment>
  <span
    class="bp3-tag bp3-intent-warning"
  >
    <span
      class="bp3-text-overflow-ellipsis bp3-fill"
    >
      404 NOT FOUND
    </span>
  </span>
</DocumentFragment>
`;

exports[`StatusTagComponent StatusTagComponent renders status 500 internal error intent danger 1`] = `
<DocumentFragment>
  <span
    class="bp3-tag bp3-intent-danger"
  >
    <span
      class="bp3-text-overflow-ellipsis bp3-fill"
    >
      500 INTERNAL ERROR
    </span>
  </span>
</DocumentFragment>
`;
